www/js/home.js

```diff
@@ -17,79 +17,132 @@ function loadHome() {
+    let templateHTML = app.templates.newsTemplate({news: news});
     $(templateHTML).hide().appendTo(newsList).fadeIn(1000);
   }
 
+  function destroyNewsInfinite() {
+    app.infiniteScroll.destroy(newsTab);
+    newsTab.find('.infinite-scroll-preloader').remove();
+  }
+
+  function loadMoreNews() {
+    $.getJSON(API + '/news?page=' + page)
+      .then(function(resp) {
+        addNews(resp.news);
+        page++;
+        loadingNews = false;
+      });
+  }
+
+  function moreNews() {
+    if (!loadingNews) {
+      if (page <= totalPages) {
+        loadingNews = true;
+        loadMoreNews();
+      } else {
+        destroyNewsInfinite();
+      }
+    }
+  }
+
   function initEventStream(events) {
-    var lastDay = null;
+    let lastDay = null;
 
     if (events.length != 0) {
-      for (var event of events) {
+      for (let event of events) {
         event.start = new Date(event.start);
         event.end = new Date(event.end);
 
-        // Add header on new days. We only download events for one week - checking date is enough
+        // Add header on new days
         if (lastDay != event.start.getDate()) {
           dayHTML = app.templates.dayTitleTemplate({date: event.start});
-          eventTab.append(dayHTML);
+          eventList.append(dayHTML);
+          // Update start date to the start date of the last event + 1 day
+          start.setDate(event.start.getDate() + 1);
         }
 
-        var templateHTML = app.templates.dayTemplate({hasEvents: true,
+        let templateHTML = app.templates.dayTemplate({hasEvents: true,
           events: [event]});
-        eventTab.append(templateHTML);
+        eventList.append(templateHTML);
         lastDay = event.start.getDate();
       }
-    } else {
-      var templateHTML = app.templates.dayTemplate({hasEvents: false});
+    }
+    if (lastEvent) {
+      let templateHTML = app.templates.dayTemplate({hasEvents: false});
       eventTab.append(templateHTML);
     }
   }
 
-  function destroyInfinite() {
-    app.infiniteScroll.destroy(newsTab);
-    newsTab.find('.infinite-scroll-preloader').remove();
+  function destroyEventInfinite() {
+    app.infiniteScroll.destroy(eventTab);
+    eventTab.find('.infinite-scroll-preloader').remove();
   }
 
-  function loadMoreNews() {
-    $.getJSON(API + '/news?page=' + page)
-      .then(function(resp) {
-
-        addNews(resp.news);
-        page++;
-        loadingNews = false;
-      });
-  }
-
-  function moreNews() {
-    if (!loadingNews) {
-      if (page <= totalPages) {
-        loadingNews = true;
-        loadMoreNews();
+  function moreEvents() {
+    if (!loadingEvents) {
+      if (!lastEvent) {
+        loadingEvents = true;
+        loadMoreEvents();
       } else {
-        destroyInfinite();
+        destroyEventInfinite();
       }
     }
   }
 
+  function loadMoreEvents() {
+    $.getJSON(API + '/events?start=' + start.yyyymmdd() + '&end=' + end.yyyymmdd())
```

Comment:
Are you downloading up to one year of events at a time?

Reply:
I think this can be rewritten in the same way as a normal ajax request where data is specified instead of building a custom URL. So essentially you can pass a json object containing start and end. 
https://api.jquery.com/jquery.getjson/

Reply:
@navro Yes, this was the only way I got it to work since I'm new to the asynchronous JS. But in practice a year full of events might be maximum 20 events. @olofenglund96 Ok, I didn't know that was possible. Maybe we should discuss this further in a few weeks?

Reply:
@davidkrantz Maybe on stage, but there are about 200 on production (from last year to now) :)

Reply:
Perheps we should wait with this and discuss it on our first meeting.

Reply:
@navro I meant that there aren't that many events from any point in time to one year later. Sure, now during the nollning there are quite a few, but during the semester there aren't that many. But yes, waiting and discussing this on the meeting sounds like a good idea.